src/validations.js

```diff
@@ -35,12 +35,16 @@ export const paramTypes = (...paramTypeList) => (fnName) => (params) => {
 
 export const sameParamType = () => (fnName) => (params) => {
```

Comment:
I think we can probably implement this way and:

```js
const nonNullParams = params.filter((param) => param.dataType !== 'null');

if (nonNullParams.length > 0) {
  nonNullParams.reduce((acc, current) => {
    if (acc.dataType !== current.dataType) {
      ArgumentError.throwWrongType(fnName, acc.dataType, current.dataType);
    }
    return current;
  });
};
```

Reply:
good shout, done